lib/gateway/Shard.js

```diff
@@ -1871,6 +1878,8 @@ class Shard extends EventEmitter {
                     */
                     super.emit("resume");
                     break;
+                } else {
+                    this._resumeURL = `${packet.d.resume_gateway_url}?v=${Constants.GATEWAY_VERSION}&encoding=${Erlpack ? "etf" : "json"}`;
```

Comment:
[zlib-stream compression handling](https://github.com/abalabahaha/eris/blob/4ea56e4/lib/Client.js#L445-L447) seems to be missed out:

```suggestion
                    this._resumeURL = `${packet.d.resume_gateway_url}?v=${Constants.GATEWAY_VERSION}&encoding=${Erlpack ? "etf" : "json"}`;
                    
                    if(this.client.options.compress) {
                        this._resumeURL += "&compress=zlib-stream";
                    }
```

Reply:
As I said, naïve